### DISCUSSION TITLE
RPi developing workflow

### DISCUSSION THREAD

**Discussion by lvmasterrj** [vicwomg/pikaraoke#437] [General]
Hey @vicwomg (and fellow developers), when you have some time, could you share your workflow for developing on the RPi?

Right now, I’m using a Windows PC with VSCode, and my workflow doesn’t seem very efficient: I write the code on my PC, and when I want to test it, I connect to my RPi via SSH. Then, I remove the file using sudo rm filename, create the file again, and paste the code into it using sudo nano filename.

Don't throw rocks at me :)

Thanks a lot!

**Comment by vicwomg** [comment]
There isn't any super good way to develop on the pi that I've found (lmk if you think of one), and TBH, I've done similar things to you in terms of manually editing files.

However in most situations I'd recommend committing your changes on the windows PC, then pulling them on the RPi to pick them up. So you'd need to configure both to point to your branch. You don't have to worry about messy commit history because I squash all merged PRs into a single commit.

Other option is to use a VS code plugin to remotely edit the code on your RPi over ssh exclusively, and log on to the pi to commit your changes back to github. But I agree it's faster to develop locally. I tend to develop locally, then test on the pi later.

**Comment by vicwomg** [comment]
btw, I appreciate your PRs and will review them once I get the time!
